### PR TITLE
EIP155, EIP160 implementation

### PR DESCRIPTION
**Pull request by zilm13**
Let's start
I don't like mess with 13/null in Transaction, but I don't want to expose 13 to the users. Maybe use null instead?
Also need more tests, will research how to adopt jsontestsuite transaction tests for it.

### CODE REVIEW

File: ethereumj-core/src/test/java/org/ethereum/jsontestsuite/suite/JSONReader.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public static String getTestBlobForTreeSha(String shacommit, String testcase){
 
     public static List<String> getFileNamesForTreeSha(String sha){
 
-        String result = getFromUrl("https://api.github.com/repos/ethereum/tests/git/trees/" + sha);
+        String result = getFromUrl("https://api.github.com/repos/ethereum/tests/git/trees/" + sha + "?recursive=1");
```
Comment (Nashatyrev) [inline ethereumj-core/src/test/java/org/ethereum/jsontestsuite/suite/JSONReader.java]: What was wrong with this?

File: ethereumj-core/src/main/java/org/ethereum/config/blockchain/Eip160HFConfig.java [hunk 2]
```diff
@@ -1,18 +1,45 @@
...
+
+    @Override
+    public Integer getChainId() {
+        return 1;
+    }
+
+    @Override
+    public boolean acceptTransactionSignature(Transaction tx) {
+        if (!super.acceptTransactionSignature(tx)) return false;
+        return getChainId().equals(tx.getChainId());
```
Comment (Nashatyrev) [inline ethereumj-core/src/main/java/org/ethereum/config/blockchain/Eip160HFConfig.java]: `        return tx.getChainId() == null || getChainId().equals(tx.getChainId());`

Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/config/blockchain/Eip160HFConfig.java]: Yeah, this one would be better.

File: ethereumj-core/src/test/java/org/ethereum/jsontestsuite/GitHubVMTest.java [hunk 3]
```diff
@@ -150,4 +160,28 @@ public void testRandomVMGitHub() throws ParseException {
...
+                );
+
+        for (String fileName : fileNames) {
+            if (!includedFiles.contains(fileName)) continue;
+            System.out.println("Running: " + fileName);
+            String json = JSONReader.loadJSONFromCommit(fileName, shacommit);
+            GitHubJSONTestSuite.runGitHubJsonVMTest(json);
+        }
+
+    }
```
Comment (Nashatyrev) [inline ethereumj-core/src/test/java/org/ethereum/jsontestsuite/GitHubVMTest.java]: All those tests are executed by GitHubVMTest. We just need to change commit hash to the right one

**Pull request merged**